app/src/main/java/com/duckduckgo/app/browser/BrowserTabFragment.kt

```diff
@@ -1596,7 +1596,7 @@ class BrowserTabFragment :
             val action: String?
 
             if (appLink.appIntent != null) {
-                val packageName = appLink.appIntent.component?.packageName ?: return
+                val packageName = appLink.appIntent!!.component?.packageName ?: return
```

Comment:
can't automatically cast after moving SpecialUrlDetector to another module

Reply:
Why not ? instead of !! ?

Reply:
Just saw the if condition before. deleted my first comment.